File: internal/indexer/testdata/js/models/coreModel.js
/** BaseEntity provides shared identity fields. */
class BaseEntity {
  constructor(id = 0) {
    this.id = id;
    this.createdAt = new Date().toISOString();
  }
}

/** CoreModel is a high-usage domain model referenced by many files.
 * Used as a fixture to verify coupling score distribution for popular symbols. */
class CoreModel extends BaseEntity {
  constructor(title, score) {
    super(0);
    this.title = title;
    this.score = score;
  }

  isValid() {
    return this.title.length > 0 && this.score >= 0;
  }

  describe() {
    return this.title;
  }
}

/** createCoreModel instantiates a CoreModel from raw fields. */
function createCoreModel(title, score) {
  return new CoreModel(title, score);
}

module.exports = { BaseEntity, CoreModel, createCoreModel };

File: internal/indexer/testdata/go/models/core_model.go
package models

// BaseEntity provides shared identity fields for all core models.
type BaseEntity struct {
	ID        int
	CreatedAt string
}

// CoreModel is a high-usage domain model referenced by many files.
// Used as a fixture to verify coupling score distribution for popular symbols.
type CoreModel struct {
	BaseEntity
	Title string
	Score float64
}

// NewCoreModel constructs a CoreModel with defaults.
func NewCoreModel(title string, score float64) *CoreModel {
	return &CoreModel{
		BaseEntity: BaseEntity{ID: 0},
		Title:      title,
		Score:      score,
	}
}

// IsValid returns true when the CoreModel fields are non-empty.
func (m *CoreModel) IsValid() bool {
	return m.Title != "" && m.Score >= 0
}

// Describe returns a short description of the model.
func (m *CoreModel) Describe() string {
	return m.Title
}
